pandas/core/ops.py

```diff
@@ -1002,16 +1015,37 @@ def dispatch_to_extension_op(op, left, right):
 
     # The op calls will raise TypeError if the op is not defined
     # on the ExtensionArray
+    # TODO(jreback)
+    # we need to listify to avoid ndarray, or non-same-type extension array
+    # dispatching
+
     if is_extension_array_dtype(left):
-        res_values = op(left.values, right)
+
```

Comment:
cc @jbrockmendel 

not real happy here because ndarrays take over the ops, but not sure what else to do

Reply:
Yah definitely not my favorite part of this diff.  Making sure I grok: this is to make it dispatched to reversed ops correctly?

Instead of trying to "trick" it into calling the reversed ops via `NotImplemented`, what if we defined `reversed_ops = {operator.add: radd, operator.sub: rsub, ...}` and then made the calls directly?  We'd have to be careful about `RecursionError`s, but it _might_ be cleaner.

Reply:
yeah this prob could use some refactoring, but left for after everything works

Reply:
Is this difference intended:

```
In [55]: s = pd.Series([1, 2, np.nan], dtype='Int64')

In [56]: a = s.values

In [57]: np.array([1, 2, 3]) + a
Out[57]: array([2, 4, nan], dtype=object)

In [58]: a + np.array([1, 2, 3]) 
Out[58]: IntegerArray([2, 4, nan], dtype='Int64')
```

(I thought the code here tried to avoid this difference, although I am not fully sure if that is needed)

Reply:
@Dr-Irv said (putting it in the right place):

@jorisvandenbossche On your example above where you ask "Is this difference intended:", there is no way to avoid this. If you do np.array([1, 2, 3]) + a, then the numpy __add__() method is called. If you did pd.Series([1,2,3]) + a in your example, it works, because then the Series method sees that the right operand is an EA, and dispatches to the reverse operator of the EA.